File: C#/3. Starbuzz (Decorator)/Classes/HouseBlend.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Starbuzz.Classes
{
    public class HouseBlend : Beverage
    {
        public HouseBlend()
        {
            _Description = "House Blend Coffee";
        }

        public override double Cost()
        {
            switch (_Size)
            {
                case Size.TALL:
                    return 0.89;
                case Size.GRANDE:
                    return 1.09;
                case Size.VENTI:
                    return 1.19;
                default:
                    return 1.0f;
            }
        }
    }
}

File: Java/3 - Starbuzz/src/main/java/com/mycompany/starbuzz/HouseBlend.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.starbuzz;

import static com.mycompany.starbuzz.Beverage.Size.GRANDE;
import static com.mycompany.starbuzz.Beverage.Size.TALL;
import static com.mycompany.starbuzz.Beverage.Size.VENTI;

/**
 *
 * @author ismas
 */
public class HouseBlend extends Beverage {
    public HouseBlend(){
        description = "House Blend Coffe";
    }
    
    @Override
    public double cost(){
        switch(this.getSize()){
            case TALL:
                return 0.89;
            case GRANDE:
                return 0.95;
            case VENTI:
                return 0.99;
            default:
                return 0.89;
        }
    }
}
